src/core/lib/attribute-transition-manager.js

```diff
@@ -0,0 +1,367 @@
+${attributeDeclarations.join('\n')}
+${uniformsDeclarations.join('\n')}
+${varyingDeclarations.join('\n')}
+
+void main(void) {
+  ${calculations.join('\n')}
+  gl_Position = vec4(0.0);
+}
+`;
+
+    const fs = `\
+#define SHADER_NAME feedback-fragment-shader
+
+#ifdef GL_ES
+precision highp float;
+#endif
+
+${varyingDeclarations.join('\n')}
+
+void main(void) {
+  gl_FragColor = vec4(0.0);
+}
+`;
+
+    if (this.model) {
+      this.model.destroy();
+    }
+
+    this.model = new Model(this.gl, {
+      id: this.id,
+      program: new ProgramTransformFeedback(this.gl, {vs, fs, varyings}),
+      geometry: new Geometry({
+        id: this.id,
+        drawMode: GL.POINTS
+      }),
+      vertexCount: 0,
+      isIndexed: true
+    });
+
+    this._updateModel(this.attributeTransitions);
+  }
+
+  // get current values of an attribute, clipped/padded to the size of the new buffer
+  _getCurrentAttributeState(transition) {
+    const {attribute, buffer, bufferSize} = transition;
+    const {value, type, size} = attribute;
+
+    const newBufferSize = value.length;
+    // Enter from 0
+    // const currentValues = new (value.constructor)(newBufferSize);
+    // No entrance transition
+    const currentValues = value.slice();
+    if (buffer) {
+      // Transfer old buffer data to the new one
+      const oldBufferData = new Float32Array(bufferSize);
+
+      buffer.getData({
+        dstData: oldBufferData,
+        srcByteOffset: 0
+      });
+
+      const len = Math.min(bufferSize, newBufferSize);
+      for (let i = 0; i < len; i++) {
+        currentValues[i] = oldBufferData[i];
+      }
+    }
+
+    return {size, type, value: currentValues};
+  }
+
+  _getTransitionSettings(transition) {
+    const {opts} = this;
+    const {accessor} = transition.attribute;
+
+    let settings = Array.isArray(accessor) ?
+      accessor.map(a => opts[a]).find(Boolean) :
+      opts[accessor];
+
+    // Shorthand: use duration instead of parameter object
+    if (Number.isFinite(settings) && settings > 0) {
+      settings = {duration: settings};
+    }
+
+    return settings && settings.duration ? {
+      duration: settings.duration,
+      easing: settings.easing || (t => t),
+      onStart: settings.onStart || noop,
+      onEnd: settings.onEnd || noop,
+      onInterrupt: settings.onInterrupt || noop
+    } : null;
+  }
+
+  // Updates transition from/to and buffer
+  _updateAnimation(transition) {
+    const {attribute, buffer} = transition;
+    const {value, size} = attribute;
+
+    const transitionSettings = this._getTransitionSettings(transition);
+
+    const needsNewBuffer = !buffer || transition.bufferSize !== value.length;
+
+    let fromState;
+    if (transitionSettings) {
+      fromState = this._getCurrentAttributeState(transition);
+    }
+    const toState = new Buffer(this.gl, {size, data: value});
+
+    if (needsNewBuffer) {
+      if (buffer) {
+        buffer.unbind();
+        buffer._deleteHandle();
```

Comment:
Looks like we need a public method in luma:buffer class.

Reply:
Actually you can just call `buffer.delete()`, it is implemented at `Resource` level.